samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c

```diff
@@ -227,6 +246,10 @@ static void cloud_lwm2m_rd_client_event_cb(struct lwm2m_ctx *client_ctx,
 	case LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE:
 		mosh_print("LwM2M: Registration complete");
 		connected = true;
+		/* Check if session lifetime needs to be updated. */
+		if (update_session_lifetime) {
+			cloud_lwm2m_rd_client_update_lifetime(client.srv_obj_inst);
```

Comment:
Actually.. why don't you call this every time on `LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_TRANSFER_COMPLETE` 

You don't need a boolean then.

Reply:
And you save one LwM2M UPDATE message, because we would use the configured lifetime already in registration phase.

Reply:
Good point, I didn't think of it when I removed the second place where this function was called from. I'll change it then.

Reply:
But it doesn't seem to work. Even though the function is called and it set the lifetime to 12 hours, the session lifetime is still 60 seconds. I guess the value is set too early and gets overwritten by the one from the server during registration?

Reply:
@SeppoTakalo I also tried calling the update function when event `LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_TRANSFER_COMPLETE` is received, but that doesn't work either. Based on my quick trials, it has to be done like it's currently implemented.

Reply:
I think this might be that you changed a lifetime of wrong server instance object. The 
During a bootstrap it is probably not selected. So I see that you suggestion seem to be the easiest one.

`cloud_lwm2m_rd_client_update_lifetime()` have no error checks, so even if its called with `-1` it silently passes.